[2021-09-14 18:50:52] [INFO] red: Loading packages...
[2021-09-14 18:50:53] [INFO] red: Loaded packages: permissions, general, admin, alias, cleanup, customcom, downloader, filter, image, mod, modlog, google, speak, todo, typeracer, clsroom
[2021-09-14 18:50:56] [INFO] red: Connected to Discord. Getting ready...
[2021-09-14 19:02:20] [CRITICAL] red.main: The main bot task didn't handle an exception and has crashed
Traceback (most recent call last):
  File "C:\Users\sugan\redenv\lib\site-packages\redbot\__main__.py", line 475, in red_exception_handler
    red_task.result()
  File "C:\Users\sugan\redenv\lib\site-packages\redbot\__main__.py", line 387, in run_bot
    await red.start(token, bot=True, cli_flags=cli_flags)
  File "C:\Users\sugan\redenv\lib\site-packages\redbot\core\bot.py", line 1202, in start
    return await super().start(*args, **kwargs)
  File "C:\Users\sugan\redenv\lib\site-packages\discord\client.py", line 666, in start
    await self.connect(reconnect=reconnect)
  File "C:\Users\sugan\redenv\lib\site-packages\discord\shard.py", line 435, in connect
    raise item.error
  File "C:\Users\sugan\redenv\lib\site-packages\discord\shard.py", line 151, in worker
    await self.ws.poll_event()
  File "C:\Users\sugan\redenv\lib\site-packages\discord\gateway.py", line 582, in poll_event
    raise ConnectionClosed(self.socket, shard_id=self.shard_id, code=code) from None
discord.errors.ConnectionClosed: Shard ID 0 WebSocket closed with 4004
[2021-09-14 19:02:21] [WARNING] red.main: Attempting to die as gracefully as possible...
[2021-09-14 19:02:21] [INFO] red.main: Shutting down from unhandled exception
[2021-09-14 19:09:38] [WARNING] red.main: Please do not use Ctrl+C to Shutdown Red! (attempting to die gracefully...)
[2021-09-14 19:09:38] [ERROR] red.main: Received KeyboardInterrupt, treating as interrupt
[2021-09-14 19:09:38] [INFO] red.main: Signals.SIGINT received. Quitting...
[2021-09-14 19:09:38] [INFO] red.main: Please wait, cleaning up a bit more
